docs/io/fits/usage/headers.rst

```diff
@@ -328,39 +332,32 @@ If a keyword contains more than 8 characters Astropy will automatically use a
 HIERARCH card, but will also issue a warning in case this is in error.
 However, one may explicitly request a HIERARCH card by prepending the keyword
 with 'HIERARCH ' (just as it would appear in the header).  For example,
-``header['HIERARCH abcdefghi']`` will create the keyword ``abcdefghi`` without
+``hdr['HIERARCH abcdefghi']`` will create the keyword ``abcdefghi`` without
 displaying a warning.  Once created, HIERARCH keywords can be accessed like any
-other: ``header['abcdefghi']``, without prepending 'HIERARCH' to the keyword.
-HIEARARCH keywords also differ from normal FITS keywords in that they are
```

Comment:
Just so I remember: Does anyone know when this changed? From my tests they are **not case-sensitive**.

```
from astropy.io import fits

hdr = fits.Header()
hdr['abcdefghijklm'] = 10
hdr['abcdefghijklm'.upper()]
hdr['HIERARCH abcdefghijklm'.upper()]
```

Reply:
@MSeifert04 - could be #5321. Not sure what's exactly the expected behavior.

Reply:
But that only upper-cases the first 9 chars (the HIERARCH-part) if I'm not mistaken.

Reply:
oups, misread the code. You're right!

Reply:
Yes HIERARCH is always uppercase, like a normal keyword, and it is case sensitive for the remaining part. But I'm not sure if the case-sensitive feature should allow to have two keywords with only the case that differs.  I did not found a detailed description, and from my experience ESO uses only uppercase tokens.